java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -1506,9 +1623,12 @@ class DatabaseImpl(
         is FieldType.Tuple ->
             throw NotImplementedError("[FieldType.Tuple]s not currently supported.")
         is FieldType.ListOf -> getTypeId(fieldType.primitiveType, database)
-        // TODO(b/144508181)
-        is FieldType.InlineEntity ->
-            throw NotImplementedError("[FieldType.InlineEntity]s not currently supported.")
+        is FieldType.InlineEntity -> {
+            var schema = requireNotNull(SchemaRegistry.getSchema(fieldType.schemaHash)) {
```

Comment:
schemas-inside-schemas as a general class are already registered so that references work properly. I didn't actually need to do anything different here, I just leverage that.